src/features/SearchPanel/SearchPanel.tsx

```diff
@@ -0,0 +1,34 @@
+"use client";
+import SearchStyle from "./SearchPanel.module.scss";
+import { SearchState } from "./Search.types";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import { getDefaultFormValues } from "./Search.utils";
+
+export function SearchPanel() {
+  const router = useRouter();
+  const handleSubmit = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+    const encodedSearch = encodeURI(formState.request);
+    router.push(`/search?q=${encodedSearch}`);
```

Comment:
Don’t redirect from the SearchPanel, it’d be better to have a common input component that doesn’t depend on where you use it